op-node/rollup/driver/sync_deriver.go

```diff
@@ -227,16 +227,27 @@ func (s *SyncDeriver) SyncStep() {
 
 	s.tryBackupUnsafeReorg()
 
-	s.Engine.TryUpdateEngine(s.Ctx)
+	// Guard: Do not send an FCU until UsafeHead is initialized.
+	// In other words, avoid sending an FCU that sets the unsafe head to genesis.
+	// This prevents side effects such as snap sync being terminated prematurely with an empty EL.
+	if s.Engine.UnsafeL2HeadInitialized() {
+		s.Engine.TryUpdateEngine(s.Ctx)
+	}
```

Comment:
This does not appear to be a guard because it does not terminate the function early. Is there meant to be a `return`?

Reply:
I ended up to remove this check because this edge case(FCUing to genesis) only happens with `l2.follow.source` which is not at this PR. Also fixed the PR description. Fixed at https://github.com/ethereum-optimism/optimism/pull/18290/commits/9e70cc2839a03dc257b88167d7288d092b525cd2